src/diagnostic.c

```diff
@@ -56,6 +56,7 @@ static const char* const diagnostic_messages[PM_DIAGNOSTIC_ID_LEN] = {
     [PM_ERR_ARGUMENT_AFTER_BLOCK]               = "unexpected argument after a block argument",
     [PM_ERR_ARGUMENT_AFTER_FORWARDING_ELLIPSES] = "unexpected argument after `...`",
     [PM_ERR_ARGUMENT_BARE_HASH]                 = "unexpected bare hash argument",
+    [PM_ERR_ARGUMENT_BLOCK_FORWARDING]          = "both a block argument and a forwarding argument; only one block is allowed",
     [PM_ERR_ARGUMENT_BLOCK_MULTI]               = "multiple block arguments; only one block is allowed",
```

Comment:
Does it make sense to use `PM_ERR_ARGUMENT_BLOCK_MULTI` message?

In all the other cases with two block arguments Prism returns `multiple block arguments; only one block is allowed`:

```ruby
foo(&a) {}

def bar(&)
  foo(&) { }
end


def bar()
  foo(&:to_s) { }
end
```

Reply:
It makes sense of course. However, I believe the detailed error message helps users, so I introduce the new error message for this.